src/api/middleware.py

```diff
@@ -63,8 +61,18 @@ def _obfuscate(self, value):
                 for i in range(len(value[key])):
                     value[key][i] = "********"
 
+    def _ignore_url_path(self, url_path):
+        return any(
+            url_path.startswith(ignore_route)
+            for ignore_route in settings.METRICS_IGNORE_ROUTES
+        )
+
     def __call__(self, request):
         url_path = request.path
+
+        if settings.DISABLE_METRICS or self._ignore_url_path(url_path):
+            return self.get_response(request)
```

Comment:
A tiny point, but maybe the `response` definition currently on 83 should be moved up to 72, and then returned here?